compiler/testSuite/01-function/assign8-v.bal

```diff
@@ -0,0 +1,18 @@
+import ballerina/io;
+
+function foo(int[] arr, int i) returns int[] {
+    arr[i] = i + 10;
+    return arr;
+}
+
+public function main() {
+    int[] arr = [];
```

Comment:
Need many more tests.  In particular, you need

_ = 1;

to check folding.

Reply:
Added